### PR TITLE
FindCommand Enhancement

### LINKED ISSUE #20: Find patients
As a nurse, I am able to search for patient by name/phone number/patient id

### PR DESCRIPTION
**Pull request by WadeForward**
Update `findCommand` such that it can takes support finding by phone number and nric.

Multiple number of phone numbers and nrics can be entered and are all case insensitive

closes #20

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -126,19 +126,20 @@ Examples:
 
 Finds patients whose names contain any of the given keywords.
```
Comment (JinHao-L) [inline docs/UserGuide.md]: Should this be updated to `Finds patients by name, NRIC or phone numbers`?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -126,19 +126,20 @@ Examples:
...
 
 * The search is case-insensitive. e.g `hans` will match `Hans`
 * The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
-* Only the name is searched.
+* Only the name, nric, and phone number are searched.
 * Only full words will be matched e.g. `Han` will not match `Hans`
 * Patients matching at least one keyword will be returned (i.e. `OR` search).
   e.g. `Hans Bo` will return `Hans Gruber`, `Bo Yang`
 * The respective appointments of patients matching at least one keyword will also be returned. 
+* The command cannot be empty, i.e. `find`. At least one among searching by name, nric, or phone number needs to be given.
```
Comment (JinHao-L) [inline docs/UserGuide.md]: Would this sound better?  
`The search requirements cannot be left empty, at least one search requirement (by name, Nric, or phone number) has to be provided`

File: docs/UserGuide.md [hunk 3]
```diff
@@ -151,7 +152,7 @@ Format: `delete INDEX`
 * Deletes the patient at the specified `INDEX`.
 * The index refers to the index number shown in the displayed patient list.
 * The index **must be a positive integer** 1, 2, 3, …​
-* Deleting a patient will also deletes all the appointments of the person.
+* Deleting a patient will also delete all the appointments of the person.
 
 Examples:
```
Comment (JinHao-L) [inline docs/UserGuide.md]: Thanks for updating the UG!

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 4]
```diff
@@ -15,22 +27,30 @@
     public static final String COMMAND_WORD = "find";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all patients whose names contain any of "
-            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
-            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " alice bob charlie";
+            + "the specified keywords (case-insensitive),"
+            + "or whose phone number or nric matches and displays them as a list with index numbers.\n"
+            + "Parameters: [" + PREFIX_NAME + "NAME [MORE_NAMES]] "
```
Comment (JinHao-L) [inline src/main/java/seedu/address/logic/commands/FindCommand.java]: Would it sound better if `and displays them as a list with index numbers` is removed?  
What do you think?

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 5]
```diff
@@ -15,22 +27,30 @@
...
+    public FindCommand(FindPatientDescriptor descriptor) {
+        this.descriptor = descriptor;
     }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
+        assert descriptor.getNamePredicate().isPresent();
+        assert descriptor.getNricPredicate().isPresent();
+        assert descriptor.getPhonePredicate().isPresent();
```
Comment (JinHao-L) [inline src/main/java/seedu/address/logic/commands/FindCommand.java]: I'm not sure about this but wouldn't the assertion fail if certain prefix is not used?  
E.g. if name prefix is unused, wouldn't there be no namePredicate?

**Review by JinHao-L — COMMENTED** [review]
Good work!
Other than the above comments, the rest LGTM 👍 

I also noticed that find_enhancement branch is branched from the fix_bug branch.  
I think it would be good to merge the fix_bug branch before this branch

File: docs/UserGuide.md (hunk 3 above)
Comment (WadeForward) [reply docs/UserGuide.md]: my pleasure 👍

File: docs/UserGuide.md (hunk 2 above)
Comment (WadeForward) [reply docs/UserGuide.md]: This is definitely better!

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 5 above)
Comment (WadeForward) [reply src/main/java/seedu/address/logic/commands/FindCommand.java]: Thanks for bring this up!

**Review by JinHao-L — APPROVED** [review]
LGTM! 👍 
Remember to update the UG (in another PR)

**Pull request merged**